### PR TITLE
Vignette of covariance adjustment with auxiliary data in a pair-matched study

### PR DESCRIPTION
**Pull request by jwasserman2**
A wish list for this vignette:

- [x] Title should heighten emphasis on concepts essential to package.  It doesn't have to satisfy a journal editor in any way, it only has to effectively market the vignette to readers who might be drawn to it.
- [x] Revise overview to emphasize propertee structures/concepts
- [x] Load MME data directly from xls, adding readxl dependency
- [x] Update clean_data.R in light of MME data intake change
- [x] script downloading data from specific URLs
- [ ] Section "External Requirements", subheaders R Packages and Data (in that order). Add readxl to R Packages as appropriate
- [ ] In "Creating the `Design` Object" section, revise to call out unit of assignment concept. (Probably also identify names `uoa()` and `cluster()`, flagging the former as less ambiguous.)
- [ ] In "Estimating..." sections, move/shorten text narrating commands to avoid having it draw reader away from reading actual commands.
- [ ] In section "Estimating Heterogeneous Intervention Effects", break up the one long paragraph and the one long code block into interleaving code blocks and narratives describing those blocks specifically.
- [ ] In section "Estimating Heterogeneous Intervention Effects", I don't see how "the second exception" (to the user experience being same as for main effects) arises in the given code block. Does it? If not, either remove the discussion under "the second exception" or indicate that that the issue doesn't arise in this example.  If so, relocate the explanation of this exception to be immediately adjacent to the code in which it occurs.
- [ ] I couldn't identify uses of the variable `county_ix` defined in the lines
```r
county_ix <- analysis1data$CONAME == coname
county_ix <- analysis2data$CONAME == coname
```
    Can it be removed?
- [ ] At end of vignette, add "This vignette has demonstrated" followed by a bulletted list of key concepts and/or commands. 
- [ ] In `get_and_clean_external_data.R`, get rid of unneeded intermediate objects when done.

### CODE REVIEW

**Comment by jwasserman2** [comment]
@josherrickson thoughts on having added `paireddata` for this vignette as a dataset for the package?
@benthestatistician thoughts on what the vignette covers?

**Comment by benthestatistician** [comment]
Re vignette coverage, I much like what I'm seeing.  I have some ideas about how the vignette could state its purposes to highlight what's special about propertee, but perhaps it would be more efficient for me to add and edit text myself sometime in the coming weeks than to start wordsmithing now. So for now I'll keep those ideas to myself. 

I figured this vignette would offer a natural opportunity to show off the `weights=ate(des)*measure_of_cluster_size` construction, so was surprised not to see it here. I'd think we should; without measures of cluster size, we estimate the effect of the treatment on school level averages.  But if we bring in cluster size as above, then we get to estimate student level effects (despite our being restricted to aggregated data).  

I'll let Josh E chime in about the paireddata addition, but I'll comment that:
-  I like having it but might suggest a name like "michigan_school_pairs"
- Would be good to say that the schools we're looking at here are the closest matches we found to the schools that had participated in an RCT in a neighboring county
- it would be worth clarifying that there was no known intervention was happening at the school pairs and triples, so the expected effects in our example are 0 (although the actual RCT it's modeled on did seem to have some effect).

**Comment by josherrickson** [comment]
No objection to adding more data, as long as it's small enough to not trigger CRAN's size warning. I believe it warns if any directory is greater than 1Mb, based upon this note on rlemon's checks:

```
 Result: NOTE
    installed size is  5.0Mb
    sub-directories of 1Mb or more:
      lemon   2.9Mb
      libs    1.9Mb
```

It wouldn't be a total stopper, but avoiding any notes should ease CRAN submissions.

**Comment by jwasserman2** [comment]
I like the name "michigan_school_pairs"; I'll re-name it. No issues about size--it's only half a kilabyte. In the vignette, I'll clarify we expect to find no effect given there was no intervention to our knowledge. I agree that you should make comments directly to the vignette, Ben.

**Comment by benthestatistician** [comment]
This vignette calls for downloading data directly from the state of MI (no permission to redistribute).  The data are distributed in xls format, at a static URL (namely, `https://www.michigan.gov/cepi/-/media/Project/Websites/cepi/MiSchoolData/historical/Historical_Assessments/2011-2014MME.zip`).

So I'd like to propose that we take on a `readxl` package dependency, if necessary, in order to incorporate downloading and cleaning directly into the vignette. 

The readxl package is part of the tidyverse, but it as [advertises itself](https://readxl.tidyverse.org/) on its lack of external dependencies, and [it doesn't import or depend on](https://cloud.r-project.org/web/packages/readxl/index.html) the whole rest of the tidyverse[^1]. It seems to be the least troublesome solution to the problem of reading Excel files, per the [wisdom of the internet](https://stackoverflow.com/questions/7049272/importing-excel-files-into-r-xlsx-or-xls). Thoughts? @josherrickson @jwasserman2 

[^1]: For me, the readxl install triggered installation of direct dependencies "cellranger" and "progress", plus indirect dependencies "rematch" (via cellranger) and "hms" (via progress).  I already had remaining direct dependencies, including "tibble" and "cpp11" (but not "covr", which readxl suggests).

**Comment by benthestatistician** [comment]
Let me retract my prior question in favor of a different one.  Suppose that after revisions of this vignette, it has a package dependency that the main package lacks as well as 2 external data dependencies, both handled with `download.file(<external URL>, tempfile)`. What more would remain to do, if anything, before including it in the package website (while excluding it from the CRAN build)?  @josherrickson

**Comment by josherrickson** [comment]
If we're not including it to CRAN, there shouldn't be any issues. The only thing I could think of is whether Github Actions (which builds the site) supports downloading something external, but I doubt that'll be an issue. We will need to add any extra packages to the GHA script: https://github.com/benbhansen-stats/propertee/blob/a8d44228f470a43857fcccb41193d4bdf09fb5e5/.github/workflows/pkgdown.yaml#L37

That just takes in a comma or newline separated list of packages, see e.g. https://github.com/benbhansen-stats/propertee/blob/a8d44228f470a43857fcccb41193d4bdf09fb5e5/.github/workflows/r.yml#L38-L43 (That was taken from optmatch, I should probably clear that list out. At some point I'll check for updates in the GHA files and get rid of those extra dependencies at that point.)

For what it's worth, I have no objection to dependencies on Vignettes which aren't included in CRAN; you can import all 21,000+ packages for all I care. I'm just vigilant to keeping out unnecessary packages from DESCRIPTION.

**Comment by benthestatistician** [comment]
@josherrickson, can you work with what's here now (in terms of bringing it into the website)?  

@jwasserman2 , I've made a number of edits and flagged some other things to do.  I'm putting this list of to-do's in to the PR statement at the top, for ease of reference.  Their overall theme is that package vignettes aren't for methodological arguments, they're for demonstrating how the package implements methodologies.  I think the vignette could better observe this rule, backgrounding some of the pitch-making (perhaps saving some of it for the methodological paper) while making more prominent the key concepts of the package.  The tasks I'll be posting aim to represent steps in this direction.  I don't see any of them as necessary to do before merging into master (while leaving the PR open).

**Comment by benthestatistician** [comment]
@josherrickson  if you see this as ready to be included in the site, when you get the chance could you merge into main and adjust as necessary for that to occur? (Ideally leaving the PR open for now.) At the moment it's even with the master branch and is passing tests.

**Comment by josherrickson** [comment]
Sorry I don't think I can keep the PR open once it's been merged. I'd argue semantically you'd want a new PR anyways.

This is up. Both title and subtitle were a bit wordy and long for a menu option, so feel free to tweak here: https://github.com/benbhansen-stats/propertee/blob/89b5e1566ec372c2e4da3aff373e34096afc291b/_pkgdown.yml#L14-L15 You can also of course place into a different position in the menu. We could also create separate menus if that's desired.

**Comment by benthestatistician** [comment]
Sorry, I should have written "if you're OK w/ merging this branch into main, please do, leaving PR open".
I've added creating a shortened title to the list of to-dos at the top of the PR.

**Comment by benthestatistician** [comment]
I've reopened #67 as a signifier that there are remaining to-do's on this vignette (enumerated in current version of PR statement). This work looks like it can be done in main, so I'm deleting the branch associated with this issue.

**Comment by benthestatistician** [comment]
I've reopened #67 as a signifier that there are remaining to-do's on this vignette (enumerated in current version of PR statement). This work looks like it can be done in main, so I'm deleting the branch associated with this issue.
